tests/MongoDB.Driver.Core.Tests/Core/Authentication/SaslPrepHelperTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿/* Copyright 2018–present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using FluentAssertions;
+using Xunit;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+namespace MongoDB.Driver.Core.Authentication
+{
+    /// <summary>
+    /// SaslPrep unit tests.
+    /// </summary>
+    public class SaslPrepHelperTests
```

Comment:
Might as well copy the tests from Java driver at com.mongodb.internal.authentication.SaslPrepTest

Reply:
Done. Definitely worth the effort, caught at least a couple of issues.